luigi/parameter.py

```diff
@@ -222,6 +257,23 @@ def parse(self, x):
         """
         return x  # default impl
 
+    def parse_list(self, xs):
+        """
+        Parse a list of values from the scheduler.
+
+        Only possible if this parameter has a batch method. This will combine the list into a single
+        parameter value using batch method.
+
+        :param xs: list of values to parse and combine
+        :return: the combined parsed values
+        """
+        if self._batch_method is None:
+            raise NotImplementedError('No batch method found')
+        elif not xs:
+            raise ValueError('Empty parameter list passed to parse_list')
+        else:
+            return self._batch_method(map(self.parse, xs))
```

Comment:
Suppose you have the following code:

```
class ExampleTask(luigi.Task):
    param = IntParameter(batch_method=list)
```

Now `ExampleTask(1).param` is an int, `ExampleTask(2).param` is an int, and so on. But if I send these both to the scheduler and get back a batch of both of them, I get `ExampleTask([1, 2])`. `ExampleTask([1, 2]).param` is a list of ints. If I call serialize on it, it's going to fail. I'll also have to write weird code that handles both ints and lists of ints.

To make this work cleanly, I'll need to change the type of this parameter from int to list of ints. I'll need to change the parse and serialize functions. Now `batch_method=list` won't work because if I batch `ExampleTask([1])` and `ExampleTask([2])`, I get `ExampleTask([[1, 2]])`. Again, the types are wrong because `list` changes the type from `X` to `list<X>`. So I also need to import or write a flatten function to return a list.

So in order to get a `list` batch method, I need to write custom parse and serialize methods and find a flatten function. The parse and serialize methods might also need to handle weird escaping in case the parameter value could include the list separator. The only way I can see to make this all easy and likely to succeed for the average user is to provide a well-tested decorator. Otherwise, they're likely to fall into the same trap you did (and I did in some of my unit tests before getting this far) and just set `batch_method=list`.